cmd/index/add/add.go

```diff
@@ -78,64 +76,33 @@ func NewIndexAddCmd(ctx context.Context, opt *options.CommonOptions) *cobra.Comm
 
 // RunIndexAdd implements the index add command.
 func (o *IndexAddOptions) RunIndexAdd(ctx context.Context, args []string) error {
+	var err error
+	var indexes *index.Config
+	var currentIndexes []config.Index
+
 	name := args[0]
-	nameYaml := fmt.Sprintf("%s%s", name, ".yaml")
 	url := args[1]
-	indexFile := filepath.Join(config.IndexesDir, nameYaml)
-
-	indexConfig, err := index.NewConfig(config.IndexesFile)
-	if err != nil {
-		return err
-	}
 
-	if _, err := indexConfig.Get(name); err == nil {
-		o.Printer.Warning.Printf("%s already exists with the same configuration, skipping\n", name)
-		return nil
-	}
-
-	remoteIndex, err := index.Fetch(ctx, url, name)
+	indexCache, err := cache.New(config.IndexesFile, config.IndexesDir)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to create index cache: %w", err)
 	}
 
-	// Save the new index.
-	err = remoteIndex.Write(indexFile)
-	if err != nil {
-		return err
+	if err = indexCache.Add(ctx, name, url); err != nil {
+		return fmt.Errorf("unable to add index: %w", err)
 	}
 
-	// Keep track of the newly created index file in indexes.yaml.
-	ts := time.Now().Format(consts.TimeFormat)
-	entry := index.ConfigEntry{
-		Name:             name,
-		AddedTimestamp:   ts,
-		UpdatedTimestamp: ts,
-		URL:              url,
+	if indexes, err = indexCache.Write(); err != nil {
+		return fmt.Errorf("unable to write cache to disk: %w", err)
 	}
 
-	indexConfig.Add(entry)
-
-	if err := indexConfig.Write(config.IndexesFile); err != nil {
-		return err
+	for _, cfg := range indexes.Configs {
+		currentIndexes = append(currentIndexes, config.Index{
+			Name: cfg.Name,
+			URL:  cfg.URL,
+		})
```

Comment:
By definition, a cache is ephemeral. So I'd expect these two steps in reverse order:
- first, store the config (no matter what will happen with the cache)
- second, store the cache (if can't store the cache is not a real problem, it can be recreated at any time).

Note:
not really important for now

Reply:
Totally agree about the cache,  still, I would prefer that the changes to the config file are done only if the run of the command is successful. The user should not expect anything if the command fails. 
Btw, I changed the logic of the config file update.